src/playlist-loader.js

```diff
@@ -104,6 +105,13 @@
 
       // update the playlist loader's state in response to a new or
       // updated playlist.
+      haveMetadataFirst = function(error, xhr, url) {
+        if (error) {
+          loader.state = 'HAVE_MASTER';
```

Comment:
We only want to reset the loader.state to 'HAVE_MASTER' if it was in the 'HAVE_MASTER' state before the request that caused the error. Otherwise, this would trigger a loadedmetadata each time we switch playlists in response to an error.